usermods/Internal_Temperature_v2/usermod_internal_temperature.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include "wled.h"
+
+class InternalTemperatureUsermod : public Usermod
+{
+
+private:
+  unsigned long loopInterval;
+  unsigned long lastTime = 0;
+  bool isEnabled;
+  float temperature = 0;
+
+  static const char _name[];
+  static const char _enabled[];
+  static const char _loopInterval[];
+
+  // any private methods should go here (non-inline methosd should be defined out of class)
+  void publishMqtt(const char *state, bool retain = false); // example for publishing MQTT message
+
+public:
+  void setup()
+  {
+  }
+
+  void connected()
+  {
+  }
+
+  void loop()
+  {
+    // if usermod is disabled or called during strip updating just exit
+    // NOTE: on very long strips strip.isUpdating() may always return true so update accordingly
+    if (!isEnabled || strip.isUpdating() || millis() - lastTime <= loopInterval)
+      return;
+
+    lastTime = millis();
+
+#ifdef ESP8266 // ESP8266
+    // does not seem possible
+    temperature = -1;
+#elif defined(CONFIG_IDF_TARGET_ESP32S2) // ESP32S2
+    temperature = -1;
+#else                                    // ESP32 ESP32S3 and ESP32C3
+    temperature = roundf(temperatureRead() * 100) / 100;
+#endif
+
+#ifndef WLED_DISABLE_MQTT
+    if (WLED_MQTT_CONNECTED)
+    {
+      char array[10];
+      snprintf(array, sizeof(array), "%f", temperature);
+      publishMqtt(array);
+    }
+#endif
+  }
+
+  void addToJsonInfo(JsonObject &root)
+  {
+    // if "u" object does not exist yet wee need to create it
+    JsonObject user = root["u"];
+    if (user.isNull())
+      user = root.createNestedObject("u");
+
+    JsonArray lightArr = user.createNestedArray(FPSTR(_name));
+    lightArr.add(temperature);
+    lightArr.add(F(" °C"));
+
+
+    // if "sensor" object does not exist yet wee need to create it
+    JsonObject sensor = root[F("sensor")];
+    if (sensor.isNull())
+      sensor = root.createNestedObject(F("sensor"));
+
+    JsonArray temp = sensor.createNestedArray(FPSTR(_name));
+    temp.add(temperature);
+    temp.add(F("°C"));
+  }
+
+  void addToJsonState(JsonObject &root)
```

Comment:
@blazoncek : better to have this methods as an "empty body", or better to delete empty methods, so the usermod will not overwrite the "upper class" default methods?

Reply:
Better to remove these and rely on virtual ones from base class.
If these were required they would be pure virtual in base class.

Reply:
Done